test/email_alert_api_test.rb

```diff
@@ -79,6 +79,48 @@
 
         assert_equal("weekly", subscription_attrs.fetch("frequency"))
       end
+
+      it "returns the subscription attributes on /latest" do
+        subscription_attrs = api_client.get_latest_matching_subscription(1)
+          .to_hash
+          .fetch("subscription")
+
+        assert_equal(1, subscription_attrs.fetch("id"))
+      end
+    end
+
+    describe "user unsubscribed then resubscribed at the same frequency" do
+      before do
+        stub_email_alert_api_has_subscriptions(
+          [1, "weekly", ended: true],
```

Comment:
What happens if the user has more than one active subscription?

Reply:
Also what happens if a user's subscriptions look like: 

```
[1, "weekly"],
 [2, "daily", ended: true]
```

Reply:
That's not true. At the moment you can create a daily and weekly etc, subscription to the same list.

Reply:
I thought one automatically unsubscribed the other?

Reply:
I thought @brucebolt  created two yesterday. It's worth checking, and adding a test if required and a commit note if not.

Reply:
Bruce & I just checked this; I signed up for Daily and then Immediately on integration (both from the `/email/subscriptions/new?topic_id=find-eu-exit-guidance-for-your-business-appear-in-find-eu-exit-guidance-business-finder` endpoint).

In the database, the Daily one was `ended` with reason `frequency_changed`, and the Immediately one has a 'source' of `frequency_changed`, so it doesn't look like it's possible to have multiple active subscriptions.

But we did look for a test and couldn't find one that captures this, so I'll add one in a PR 👍

Reply:
Oh actually it is being tested here: https://github.com/alphagov/email-alert-api/blob/8c68924f632760c22154757e54b903841fb7a86d/spec/integration/create_subscription_spec.rb#L35-L58